src/Entity/AdvancedShipmentMetadataAwareTrait.php

```diff
@@ -18,12 +18,16 @@
 trait AdvancedShipmentMetadataAwareTrait
 {
     /**
-     * @ORM\Column(name="advanced_shipping_metadata", type="array")
+     * @ORM\Column(name="advanced_shipping_metadata", type="array", nullable=true)
      */
-    private array $advancedShippingMetadata = [];
+    private ?array $advancedShippingMetadata = [];
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** don't we have a problem with the setter passing it the value `null`?‏

Reply:
I casted the values to have array